runtime/bcutil/ClassFileWriter.cpp

```diff
@@ -310,6 +310,9 @@ ClassFileWriter::analyzeConstantPool()
 		case J9CPTYPE_ANNOTATION_UTF8:
 			addEntry(J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *) cpItem), i, CFR_CONSTANT_Utf8);
 			break;
+		case J9CPTYPE_CONSTANT_DYNAMIC:
+			addNASEntry(NNSRP_GET(((J9ROMConstantDynamicRef *) cpItem)->nameAndSignature, struct J9ROMNameAndSignature*));
```

Comment:
Can this use the `J9ROMCONSTANTDYNAMICREF_NAMEANDSIGNATURE(base)` macro from j9nonbuilder.h?

Reply:
updated